src/core/compact_object.h

```diff
@@ -185,13 +161,14 @@ class CompactObj {
     CompactObj res;
     memcpy(&res.u_, &u_, sizeof(u_));
     res.taglen_ = taglen_;
-    res.mask_ = mask_ | REF_BIT;
+    res.mask_ = mask_;
+    res.mask_bits_.ref = 1;
```

Comment:
`mask_` and `mask_bits_` are both part of the same union. What you do here is:

1. Set `res.mask_` field to active and assign to `mask_`
2. Set `res.mask_bits_.ref = 1` which changes the active field of the union to `mask_bits_`. 

After this any access to `res.mask_` is ub.

Reply:
So any caller of `AsRef` here will get UB if they use any member function that uses `res.mask_` without noticing.

Reply:
All these bits can be active at the same time.

It's not UB

Reply:
I am confused. Both `mask_` (plain uint) and `mask_bits_`  (unamed struct) are `different fields` of the `same unamed union`.  Unions can have one active field at a time ?